src/xdp/rx.c

```diff
@@ -1228,6 +1292,10 @@ XdpRxQueueDereference(
     if (XdpDecrementReferenceCount(&RxQueue->ReferenceCount)) {
         TraceInfo(TRACE_CORE, "Deleting RxQueue=%p", RxQueue);
         XdpIfDeregisterClient(RxQueue->Binding, &RxQueue->BindingClientEntry);
+        if (RxQueue->PcwInstance != NULL) {
+            PcwCloseInstance(RxQueue->PcwInstance);
+            RxQueue->PcwInstance = NULL;
+        }
         ExFreePoolWithTag(RxQueue, XDP_POOLTAG_RXQUEUE);
```

Comment:
Is it worth putting this all in a separate function now? It's kind of grown a bit over time...

Reply:
I think we agreed within our team that putting the cleanup inside the dereference routine is ideal, if that's the question?